execution_engine/src/core/engine_state/engine_config.rs

```diff
@@ -72,4 +57,99 @@ impl EngineConfig {
     pub fn system_config(&self) -> &SystemConfig {
         &self.system_config
     }
+
+    /// Get the engine config's chain kind.
+    #[must_use]
+    pub fn chain_kind(&self) -> ChainKind {
+        self.chain_kind
+    }
+}
+
+/// This is a builder pattern applied to the [`EngineConfig`] structure to shield any changes to the
+/// constructor, or contents of it from the rest of the system.
+///
+/// Any field that isn't specified will be defaulted.
+#[derive(Default, Debug)]
+pub struct EngineConfigBuilder {
+    max_query_depth: Option<u64>,
+    max_associated_keys: Option<u32>,
+    max_runtime_call_stack_height: Option<u32>,
+    wasm_config: Option<WasmConfig>,
+    system_config: Option<SystemConfig>,
+    chain_kind: Option<ChainKind>,
+}
```

Comment:
Why not add the new field to the `new` constructor? I don't think that using the defaults for struct like this is a good idea, seems like an easy way to make mistakes.